### PR TITLE
feat(ZENKO-1336): add rclone submodule

### PR DESCRIPTION
**Pull request by giacope**
**What does this PR do, and why do we need it?**

This PR adds rclone as a submodule. The Dockerfile contains the necessary steps to   patch rclone to our needs.

### CODE REVIEW

**Comment by ssalaues** [comment]
@giacomoguiulfo I think the difference is negligible especially with compression. Plus with the current Dockerfile, we could keep it on Docker automated builds.

**Comment by giacope** [comment]
@ssalaues Updated to use busybox instead of scratch.

**Review by ssalaues — APPROVED** [review]

**Review by tmacro — APPROVED** [review]

**Comment by giacope** [comment]
@bert-e approve

**Comment by NicolasT** [comment]
If you do patches, why use a submodule?

Either use a subtree, patch in-tree and periodically sync, or simply have a Dockerfile which downloads the source tree, applies patches, builds the binary, no submodule required.

**Comment by ssalaues** [comment]
@NicolasT I like your Dockerfile idea the best. Keeps it simple and removes need to deal with submodules.

File: images/rclone/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+COPY patches /go/src/github.com/ncw/rclone/patches
+WORKDIR /go/src/github.com/ncw/rclone/
+
+RUN for file in patches/*.diff ; do \
+        patch -p 1 < "$file" || exit 1 ; \
+    done && \
+    CGO_ENABLED=0 go build -a -installsuffix cgo && \
+    mv rclone /bin/rclone
+
+FROM golang:alpine AS dev
```
Comment (NicolasT) [inline images/rclone/Dockerfile:13]: What's the use of this intermediate containers?

Comment (giacope) [reply images/rclone/Dockerfile:13]: Build provides an environment to build the binary.

Dev's purpose is to have a lightweight version of build for testing and debugging purposes. It's not intermediate. It's a different target.

Comment (NicolasT) [reply images/rclone/Dockerfile:13]: Ah, I wasn't aware multi-stage builds got each stage tagged and kept for later execution.

**Pull request merged**